WordPress/src/main/java/org/wordpress/android/viewmodel/mlp/ModalLayoutPickerViewModel.kt

```diff
@@ -61,15 +61,19 @@ class ModalLayoutPickerViewModel @Inject constructor(
     private val _onCreateNewPageRequested = SingleLiveEvent<PageRequest.Create>()
     val onCreateNewPageRequested: LiveData<PageRequest.Create> = _onCreateNewPageRequested
 
-    private val site: SiteModel by lazy {
-        requireNotNull(siteStore.getSiteByLocalId(selectedSiteRepository.getSelectedSiteLocalId()))
+    private val site: SiteModel? by lazy {
+        siteStore.getSiteByLocalId(selectedSiteRepository.getSelectedSiteLocalId())
```

Comment:
1. Praise (❤️): Making the `SiteModel` nullable is definitely the right way to avoid the crash that is happening, especially since the `siteStore.getSiteByLocalId(...)` can return a nullable value. Thanks for making this improvement.
2. History (⏲️): This piece of code was updated as part of this [refactor PR](https://github.com/wordpress-mobile/WordPress-Android/pull/15197), which did a major change to using the `Repository` in favor of the `Prefs` in order to getting a `Selecting Site`, for at least most of the cases. In this instance, the previous logic was using the `Prefs` to do so, but in this PR it was updated to use the `Repository` instead.
3. Suggestion (💡): By making the `SiteModel` nullable, this will solve the crashed, but I wonder whether showing the `Error Screen` instead is better, as the users will not be able to recover from that anyway and thus will not be able to use the `Layout Picker`. I am also seeing that these crashes started happening from the `20th/21st of September 2021`, which is when [18.2](https://github.com/wordpress-mobile/WordPress-Android/releases/tag/18.2) was released, which included the [refactor PR](https://github.com/wordpress-mobile/WordPress-Android/pull/15197) I explained above, which started causing these crashes. Thus, my suggestion would be to maybe update this to `selectedSiteRepository.getSelectedSiteLocalId(true)` in order to revert getting the `Selected Repository` from the `Prefs`. I am still not sure in which situation and why getting it from `Prefs` doesn't crash the app, since it is almost guaranded that when the user enters the `Layout Picker` screen, this `Selected Repository` value will be present in the `Repository` cache. However, maybe something is clearing that cache and getting it from `Prefs` makes it more robust. I am not sure... Maybe this whole flows needs a bit of additional investigating to understand what is what. 🤔

Reply:
Thank you for the prompt review @ParaskP7. I really appreciate this 🙇 

> I wonder whether showing the Error Screen instead is better, as the users will not be able to recover from that anyway and thus will not be able to use the Layout Picker

I haven't been able to trigger this behaviour in real conditions but setting a `null` value for the site model should result in an error screen where the user is able to retry (this won't actually work), proceed with a blank page or close the screen.
<details>
<summary>screenshot</summary>

![device-2021-10-05-154801](https://user-images.githubusercontent.com/304044/136027818-ec11541b-5caf-4c40-8959-79a839efc213.png)
</details>

> I am also seeing that these crashes started happening from the 20th/21st of September 2021, which is when 18.2 was released

It seems that with 18.2 the issue became more common but this started to happen since `17.8` ([e.g.](https://sentry.io/organizations/a8c/issues/2550920306/events/b2eb6b0500924a52b781c68fd8de7946/?project=1438088)) 🤔 

> my suggestion would be to maybe update this to selectedSiteRepository.getSelectedSiteLocalId(true)

I agree with that. I changed it with 7871b05dd010f50dc71a4be3846b033a1276d4d9

> Consider making the SiteModel site within the SiteUtils.isBlockEditorDefaultForNewPost(...) method explicitly @Nullable 

Nice suggestion 👍 . Applied with 1f8f1005aa5adf0ab228977c36c51dd6ffb29144

I also added a release note with 391efede26d747ce1a5714050348031478ae7f27

Reply:
👋 @antonis !

Thanks so much for the updates and all your thoughts! 🥇 

> It seems that with 18.2 the issue became more common but this started to happen since 17.8 (e.g.) 🤔

Hmmm, seems weird, let's keep an eye on this. I wonder how can we track this going forward, maybe through the analytics that exist on the `Layout Picker`, at least once a while, and see if the error screen shown increases as time passes-by.

Reply:
> I wonder how can we track this going forward, maybe through the analytics that exist on the Layout Picker, at least once a while, and see if the error screen shown increases as time passes-by.

That's a great idea @ParaskP7. I'll add a reminder to check on this.

Reply:
You are the best Mr. Antonis! ❤️